src/Raven.Server/Documents/Queries/FieldsToFetch.cs

```diff
@@ -36,13 +36,13 @@ public sealed class FieldsToFetch
 
         public readonly ProjectionOptions Projection;
 
-        public FieldsToFetch(IndexQueryServerSide query, IndexDefinitionBaseServerSide indexDefinition, IndexType indexType, SearchEngineType searchEngineType = SearchEngineType.None)
+        public FieldsToFetch(IndexQueryServerSide query, IndexDefinitionBaseServerSide indexDefinition, IndexType indexType, SearchEngineType searchEngineType)
         {
             Projection = new ProjectionOptions(query);
-            Fields = GetFieldsToFetch(query.Metadata, query.ProjectionBehavior, indexDefinition, searchEngineType, indexType, out AnyExtractableFromIndex, out bool extractAllStoredFields, out SingleBodyOrMethodWithNoAlias, out AnyTimeSeries);
```

Comment:
So now we don't need `searchEngineType` at all? 

I was thinking about handling it this way (if possible):

> can we retrieve that field internally for Corax in order to do highlighting but do not return in in projection results since it was not requested?

So for that we'd need to know the engine type. 

https://github.com/ravendb/ravendb/pull/17435#discussion_r1343587184

Reply:
No, that would cause the returning of the projection result, which we don't want. It is proving much more difficult than expected to obtain the same behavior.